test_conformance/extensions/cl_khr_semaphore/test_semaphores_negative_release_retain.cpp

```diff
@@ -0,0 +1,79 @@
+//
+// Copyright (c) 2024 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "semaphore_base.h"
+
+#include "harness/errorHelpers.h"
+#include <chrono>
+#include <system_error>
+#include <thread>
+#include <vector>
+
+namespace {
+
+// sema_object is not a valid semaphore object
+
+struct ReleaseInvalidSemaphore : public SemaphoreTestBase
+{
+    ReleaseInvalidSemaphore(cl_device_id device, cl_context context,
+                            cl_command_queue queue)
+        : SemaphoreTestBase(device, context, queue)
+    {}
+
+    cl_int Run() override
+    {
+        // Release invalid semaphore
+        cl_int err = CL_SUCCESS;
+        err = clReleaseSemaphoreKHR(nullptr);
+        test_failure_error(err, CL_INVALID_SEMAPHORE_KHR,
```

Comment:
Minor: this will print "expected (unknown)" because harness/errorHelpers.cpp does not know that CL_INVALID_SEMAPHORE_KHR exists.